src/AspNet.Security.OpenId/OpenIdAuthenticationHandler.cs

```diff
@@ -323,6 +330,15 @@ protected override async Task HandleChallengeAsync(AuthenticationProperties prop
                     value: string.Join(",", Options.Attributes.Select(attribute => attribute.Key)));
             }
 
+            foreach (var prop in properties.Parameters)
+            {
+                if (prop.Value is string && message.GetParameter(prop.Key) == null)
```

Comment:
If you did this

```csharp
if (prop.Value is string valueString && message.GetParameter(prop.Key) == null)
```

Then you can change it below to:

```csharp
message.SetParameter(prop.Key, valueString);
```

Reply:
My comment below addresses why I could not use SetParameter due to it prefixing my key.

Reply:
That's not what I meant.

You test if it's a string with `is string` and then call `ToString()` on it if it is. If you use the newer is syntax which assigns the value of the `is` test into a variable, you can just add that without needing the `ToString()` call.